client/post-editor/media-modal/image-editor/image-editor-canvas.jsx

```diff
@@ -67,78 +82,136 @@ const MediaModalImageEditorCanvas = React.createClass( {
+
+		MediaUtils.canvasToBlob( newCanvas, callback, this.props.mimeType, 1 );
 	},
 
 	drawImage() {
 		if ( ! this.image ) {
 			return;
 		}
 
-		if ( this.state.canvasWidth !== this.image.width ||
-			this.state.canvasHeight !== this.image.height ) {
-			this.setState( {
-				canvasWidth: this.image.width,
-				canvasHeight: this.image.height
-			} );
-		}
+		const canvas = ReactDom.findDOMNode( this.refs.canvas );
+		const imageWidth = this.image.width;
+		const imageHeight = this.image.height;
+		const transform = this.props.transform;
+		const rotated = transform.degrees % 180 !== 0;
+
+		//make sure the canvas draw area is the same size as the image
+		canvas.width = rotated ? imageHeight : imageWidth;
+		canvas.height = rotated ? imageWidth : imageHeight;
 
-		const canvas = ReactDom.findDOMNode( this.refs.canvas ),
-			context = canvas.getContext( '2d' );
+		const context = canvas.getContext( '2d' );
 
 		context.clearRect( 0, 0, canvas.width, canvas.height );
 		context.save();
 
-		//setTransform() could be replaced with translate(), but it leads to
-		//a false positive warning from eslint rule wpcalypso/i18n-no-variables
+		// setTransform() could be replaced with translate(), but it leads to
+		// a false positive warning from eslint rule wpcalypso/i18n-no-variables
 		context.setTransform( 1, 0, 0, 1, canvas.width / 2, canvas.height / 2 );
 
-		context.scale( this.props.scaleX, this.props.scaleY );
-		context.rotate( this.props.degrees * Math.PI / 180 );
-		context.drawImage( this.image, -this.image.width / 2, -this.image.height / 2 );
+		context.scale( transform.scaleX, transform.scaleY );
+		context.rotate( transform.degrees * Math.PI / 180 );
+
+		context.drawImage( this.image, -imageWidth / 2, -imageHeight / 2 );
+
 		context.restore();
 	},
 
+	updateCanvasPosition() {
+		const canvas = ReactDom.findDOMNode( this.refs.canvas );
+		const canvasX = - 50 * this.props.crop.widthRatio - 100 * this.props.crop.leftRatio;
+		const canvasY = - 50 * this.props.crop.heightRatio - 100 * this.props.crop.topRatio;
+
+		this.props.setImageEditorCropBounds(
+			canvas.offsetTop - canvas.offsetHeight * -canvasY / 100,
+			canvas.offsetLeft - canvas.offsetWidth * -canvasX / 100,
+			canvas.offsetTop + canvas.offsetHeight * ( 1 + canvasY / 100 ),
+			canvas.offsetLeft + canvas.offsetWidth * ( 1 + canvasX / 100 ) );
+	},
+
+	preventDrag( event ) {
+		event.preventDefault();
+		return false;
+	},
+
+	renderCrop() {
+		if ( ! this.props.src ) {
+			return;
+		}
+
+		return ( <Crop /> );
+	},
+
 	render() {
-		const rotatedMod = this.props.degrees % 180,
-			width = rotatedMod === 0 ? this.state.canvasWidth : this.state.canvasHeight,
-			height = rotatedMod === 0 ? this.state.canvasHeight : this.state.canvasWidth;
+		const canvasX = - 50 * this.props.crop.widthRatio - 100 * this.props.crop.leftRatio;
+		const canvasY = - 50 * this.props.crop.heightRatio - 100 * this.props.crop.topRatio;
+
+		const canvasStyle = {
+			top: 50 + '%',
```

Comment:
Are here `50` and `'%'` separate for consistency with the lines below or is there some other reason? `50%` would probably work, too.